internal/app/synchronizer/synchronizer_service.go

```diff
@@ -82,17 +86,51 @@ func (s *synchronizerService) Register(ctx context.Context, req *ipb.RegisterReq
 // added to the list of proposals to be evaluated in the current cycle. At the
 //  end of the cycle, the user defined evaluation method is triggered and the
 // matches accepted by it are returned as results.
-func (s *synchronizerService) EvaluateProposals(ctx context.Context, req *ipb.EvaluateProposalsRequest) (*ipb.EvaluateProposalsResponse, error) {
+func (s *synchronizerService) EvaluateProposals(stream ipb.Synchronizer_EvaluateProposalsServer) error {
+	var id string
+	var proposals = []*pb.Match{}
+	for {
+		req, err := stream.Recv()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		id = req.GetId()
```

Comment:
Could we just fetch the ID for the first proposal - and if the id ever changes, then we throw an error? Right now you ll pick the last one. It was not a failure scenario for non streaming solution - but for this one, we do expect all IDs to be the exact same and if they are not, rather fail than use the last one

Reply:
I meant more like:
if req.GetId() == "" --> fail // Request cannot have empty id
if id == ""  --> id == req.GetId() // Fetching the id from first object as the id you care about
if id != req.GetId() --> error // making sure every other object has the same id